api/src/controllers/changes.js

```diff
@@ -1,578 +1,396 @@
-          feed.res.write(error(503, 'Error processing your changes'));
-        } else {
-          const malformedResponse = responses.find(response => {
-            if (!response || !response.results) {
-              // See: https://github.com/medic/medic-webapp/issues/3099
-              // This should never happen, but apparently it does sometimes.
-              // Attempting to log out the response usefully to see what's occuring
-              console.error('No _changes error, but malformed response:', JSON.stringify(responses));
-              return true;
-            }
-          });
-
-          const changes = !malformedResponse &&
-              responses.length === 1 ? responses[0] : mergeChangesResponses(responses);
-
-          if (changes) {
-            log('changes merged successfully.  Response will be ended after prepareResponse() returns.');
-            prepareResponse(feed, changes);
-          } else {
-            log('Malformed response received from upstream changes request.  Response will be ended.');
-            feed.res.write(error(503, 'No _changes error, but malformed response.'));
-          }
-          feed.res.end();
-          endTimer(`getChanges():${upstreamBatchId}.end`, startTime);
-        }
-      });
-    }
-  );
+const canceledResults = responses => {
+  return responses.find(response => response && response.status === 'cancelled');
 };
 
-const mergeChangesResponses = responses => {
-  let lastSeqNum = 0;
-  let lastSeq;
-  const results = [];
-
-  responses.forEach(changes => {
-    results.push(changes.results);
-    const numericSeq = numericSeqFrom(changes.last_seq);
-    if (numericSeq > lastSeqNum) {
-      lastSeq = changes.last_seq;
-      lastSeqNum = numericSeq;
+const mergeResults = responses => {
+  let results = [];
+  responses.forEach(response => {
+    results.push.apply(results, response.results);
+  });
+  results.forEach((change, idx) => {
+    // tombstone changes are converted into their corresponding doc changes
+    if (tombstoneUtils.isTombstoneId(change.id)) {
+      results[idx] = tombstoneUtils.generateChangeFromTombstone(change);
     }
   });
 
-  const merged = Array.prototype.concat(...results); // flatten the result sets
-  merged.sort((a, b) => numericSeqFrom(a.seq) - numericSeqFrom(b.seq));
+  return results;
+};
 
-  return {
-    last_seq: lastSeq,
-    results: merged
-  };
+const writeDownstream = (feed, content, end) => {
+  if (feed.res.finished) {
+    return;
+  }
+  feed.res.write(content);
+  if (end) {
+    feed.res.end();
+  }
 };
 
-const numericSeqFrom = seq => typeof seq === 'number' ? seq : Number.parseInt(seq.split('-')[0]);
+const getChanges = feed => {
+  // impose limit to insure that we're getting all changes available for each of the requested doc ids
+  const options = {
+    since: feed.req.query && feed.req.query.since || 0,
+    limit: MAX_DOC_IDS
+  };
+  const queryOpts = ['style', 'conflicts', 'seq_interval'];
```

Comment:
Inline this constant in the line below instead of making an array then destructuring it immediately.